handlers/validator.go

```diff
@@ -412,6 +412,7 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 		validatorPageData.IncomeTotalFormatted = earnings.TotalFormatted
 		validatorPageData.IncomeToday = earnings.IncomeToday
 		validatorPageData.ValidatorProposalData = earnings.ProposalData
+		validatorPageData.FutureDutiesEpoch = protomath.MaxU64(validatorPageData.FutureDutiesEpoch, earnings.ProposalData.LastScheduledSlot/data.ChainConfig.SlotsPerEpoch)
```

Comment:
No, because the go methods are called async and line 670 could be faster.

Reply:
Right. It is possible that both go-routines access `validatorPageData.FutureDutiesEpoch` at the same time which may lead to data corruption or crashes (afaik).

I suggest the following: Add two variables in the scope of `func Validator` here: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/b03ce154335997cba7cc60f9ff24042536e719ea/handlers/validator.go#L81-L82
Something like
```
validatorPageData := types.ValidatorPageData{}

lastProposalEpoch := uint64(0) // NEW
lastSyncDutyEpoch := uint64(0) // NEW
```
Use both variables instead of accessing `validatorPageData.FutureDutiesEpoch`.

Set `validatorPageData.FutureDutiesEpoch` exactly once here
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/b03ce154335997cba7cc60f9ff24042536e719ea/handlers/validator.go#L793
like this
```
validatorPageData.FutureDutiesEpoch = protomath.MaxU64(lastProposalEpoch, lastSyncDutyEpoch)
```

Reply:
refactored accordingly